api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -593,6 +599,71 @@ public void testMe_removeAllInstitutionalAffiliations() throws Exception {
     assertThat(result.getInstitutionalAffiliations().size()).isEqualTo(0);
   }
 
+  @Test
+  public void updateContactEmail_forbidden() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(new Timestamp(new Date().getTime()));
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail(user.getContactEmail());
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
+  }
+
+  @Test
+  public void updateContactEmail_badRequest() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(null);
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail("bad email address *(SD&(*D&F&*(DS ");
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+  }
+
+  @Test
+  public void updateContactEmail_OK() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(null);
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail(user.getContactEmail());
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+  }
+
+  @Test
+  public void resendWelcomeEmail_messagingException() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    when(directoryService.resetUserPassword(anyString())).thenReturn(googleUser);
+    doThrow(new MessagingException("exception")).when(mailService).sendWelcomeEmail(any(), any(), any());
+    ResendWelcomeEmailRequest request = new ResendWelcomeEmailRequest();
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.resendWelcomeEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void resendWelcomeEmail_OK() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    when(directoryService.resetUserPassword(anyString())).thenReturn(googleUser);
+    doNothing().when(mailService).sendWelcomeEmail(any(), any(), any());
```

Comment:
Should this and the above test be verifying whether the call happened?

Reply:
Unresolved

Reply:
Addressed below